tests/integration.test.js

```diff
@@ -0,0 +1,81 @@
+const path = require('path');
+
+const webdriver = require('selenium-webdriver');
+const { Context } = require('selenium-webdriver/firefox');
+
+// The geckodriver package downloads and installs geckodriver for us.  We use
+// it by requiring it.
+require('geckodriver');
+
+const { ALL_VERSIONS } = require('./helpers');
+
+describe(__filename, () => {
+  jest.setTimeout(20000);
+
+  it('should match the Firefox implementation', async () => {
+    const driver = new webdriver.Builder().forBrowser('firefox').build();
+    driver.setContext(Context.CHROME);
+
+    function runComparisonInFirefox(args) {
+      const { FileUtils } = Components.utils.import(
+        'resource://gre/modules/FileUtils.jsm'
+      );
+      const { NetUtil } = Components.utils.import(
+        'resource://gre/modules/NetUtil.jsm'
+      );
+
+      const allVersions = args[1];
+      const callback = args[args.length - 1];
+      const file = new FileUtils.File(args[0]);
+      const scriptUri = NetUtil.newURI(file);
+
+      Services.scriptloader.loadSubScript(scriptUri.spec, this);
+
+      const results = [];
+      for (const versions of allVersions) {
+        const [a, b] = versions;
+
+        let firefoxValue = Services.vc.compare(a, b);
+
+        // Normalize values to only have -1, 1 or 0.
+        if (firefoxValue < 0) {
+          firefoxValue = -1;
+        } else if (firefoxValue > 0) {
+          firefoxValue = 1;
+        }
+
+        results.push({
+          versions,
+          mozCompare: mozCompare(a, b),
+          firefox: firefoxValue,
+        });
+      }
+
+      callback(results);
+    }
+
+    const results = await driver.executeAsyncScript(
+      `(${runComparisonInFirefox})(arguments)`,
+      path.join(__dirname, '..', 'src', 'index.js'),
+      ALL_VERSIONS
+    );
+
+    expect.assertions(ALL_VERSIONS.length + 1);
+    // Make sure we actually have versions.
+    expect(ALL_VERSIONS).not.toHaveLength(0);
+
+    for (const result of results) {
+      // We pass `versions` so that it is displayed in the output in case of an
+      // error.
+      expect({
+        versions: result.versions,
+        value: result.mozCompare,
+      }).toEqual({
+        versions: result.versions,
+        value: result.firefox,
+      });
+    }
+
+    return driver.quit();
+  });
+});
```

Comment:
it is not a super big deal, but I was thinking that instead of loading the helpers.js file in the Firefox chrome privileged global, maybe we could opt to use a strategy similar to the one you are using in the version of the unit test from this same PR.

Using a strategy like the one in the unit test, I think we may just have to call `Services.vc.compare` on the Firefox side and then compare the result with what we would get from `mozCompare`, which should also make us able to make this test quite shorter than this version.

The the version I have in mind would basically look more or less like this:

```js
  let driver;

  beforeAll(() => {
    driver = new webdriver.Builder().forBrowser('firefox').build();
    driver.setContext(Context.CHROME);
  });

  afterAll(() => {
    return driver.quit();
  });

  it('should have versions to test', () => {
    expect(ALL_VERSIONS).not.toHaveLength(0);
  });

  it.each(ALL_VERSIONS)('match Firefox Services.vc.compare("%s", "%s")', async (a, b) => {
    const fxVcCompare = await driver.executeScript(`
      return Services.vc.compare("${a}", "${b}");
    `);
    expect(fxVcCompare).toEqual(mozCompare(a,b));
  });
```

how that sounds to you? do we want to give it a try?

Reply:
ah yes, that sounds good.